service/controller/machinepool/handler/migration/machinepool_test.go

```diff
@@ -0,0 +1,294 @@
+package migration
+
+import (
+	"context"
+	"strconv"
+	"testing"
+
+	oldcapiexpv1alpha3 "github.com/giantswarm/apiextensions/v6/pkg/apis/capiexp/v1alpha3"
+	"github.com/giantswarm/micrologger/microloggertest"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	capiexp "sigs.k8s.io/cluster-api/exp/api/v1beta1"
+
+	"github.com/giantswarm/azure-operator/v5/service/unittest"
+)
+
+func TestEnsureCreatedMachinePoolIsCorrect(t *testing.T) {
```

Comment:
Test that checks if new MachinePool CR is correct. It uses golden files for expected CRs data.